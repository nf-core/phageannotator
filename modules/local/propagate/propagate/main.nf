process PROPAGATE_PROPAGATE {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/vrhyme:1.1.0--pyhdfd78af_1' :
        'biocontainers/vrhyme:1.1.0--pyhdfd78af_1' }"

    input:
    tuple val(meta) , path(reads)
    tuple val(meta2), path(fasta)
    tuple val(meta3), path(provirus_coords)

    output:
    tuple val(meta), path("${prefix}_propagate/${prefix}_propagate.tsv")    , emit: results
    tuple val(meta), path("${prefix}_propagate/${prefix}_propagate.log")    , emit: log     , optional: true
    path  "versions.yml"                                                    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    propagate.py \\
        -f ${fasta} \\
        -v ${provirus_coords} \\
        -r ${reads} \\
        -o ${prefix}_propagate \\
        -t ${task.cpus} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        propagate: \$(propagate.py --version 2>&1 | sed 's/^.*PropagAtE v//; s/ .*\$//')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir -p ${prefix}_propagate
    touch ${prefix}_propagate/${prefix}_propagate.tsv
    touch ${prefix}_propagate/${prefix}_propagate.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        propagate: \$(propagate.py --version 2>&1 | sed 's/^.*PropagAtE v//; s/ .*\$//')
    END_VERSIONS
    """
}
