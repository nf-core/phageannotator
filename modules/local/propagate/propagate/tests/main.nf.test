nextflow_process {

    name "Test Process: PROPAGATE_PROPAGATE"
    script "../main.nf"
    process "PROPAGATE_PROPAGATE"


    test("fastq.gz + fasta + provirus_coords.tsv") {
        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        [
                            file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz", checkIfExists: true),
                            file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz", checkIfExists: true)
                        ]
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id: 'test' ],
                        file(params.pipelines_testdata_base_path + "modules/local/propagate/b_fragilis_propagate_contigs.fasta", checkIfExists: true)
                    ]
                )
                input[2] = Channel.of(
                    [
                        [ id: 'test' ],
                        file(params.pipelines_testdata_base_path + "modules/local/propagate/b_fragilist_provirus_coords.tsv", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                    process.out.results,
                    path(process.out.log[0][1]).readLines()[17..32],
                    process.out.versions
                    ).match()
                }
            )
        }
    }

    test("fastq.gz + fasta + provirus_coords.tsv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        [
                            file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz", checkIfExists: true),
                            file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz", checkIfExists: true)
                        ]
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id: 'test' ],
                        file(params.pipelines_testdata_base_path + "modules/local/propagate/b_fragilis_propagate_contigs.fasta", checkIfExists: true)
                    ]
                )
                input[2] = Channel.of(
                    [
                        [ id: 'test' ],
                        file(params.pipelines_testdata_base_path + "modules/local/propagate/b_fragilist_provirus_coords.tsv", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
