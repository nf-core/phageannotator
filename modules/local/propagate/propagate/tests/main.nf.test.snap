{
    "fastq.gz + fasta + provirus_coords.tsv": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test_propagate.tsv:md5,f5dc748967a1840bff9c504804d4b6d4"
                ]
            ],
            [
                "Results file:      test_propagate.tsv",
                "Active prophages:  0",
                "",
                "                                                               ##",
                "                                                             ##  ##",
                "                                                           ##      ##",
                "######   ##  ##     ##     #######   ######    #####       ##      ##",
                "##  ##   ##  ##   ##  ##   ##        ##       ##             ##  ##",
                "######   ######   ######   ##  ###   ######    ###             ##",
                "##       ##  ##   ##  ##   ##   ##   ##           ##           ##",
                "##       ##  ##   ##  ##   #######   ######   #####            ##",
                "                                                            #  ##  #",
                "                                                           # # ## # #",
                "                                                          #   #  #   #",
                "                                                         #            #",
                ""
            ],
            [
                "versions.yml:md5,dfe5c94afe825f0011160109ed5f8bb1"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T15:12:59.933317207"
    },
    "fastq.gz + fasta + provirus_coords.tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_propagate.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_propagate.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,dfe5c94afe825f0011160109ed5f8bb1"
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test_propagate.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test"
                        },
                        "test_propagate.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dfe5c94afe825f0011160109ed5f8bb1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T15:13:10.985670747"
    }
}